temba/channels/types/instagram/views.py

```diff
@@ -0,0 +1,249 @@
+
+                if page_access_token == "":  # pragma: no cover
+                    raise Exception("Empty page access token!")
+
+                url = f"https://graph.facebook.com/v12.0/{page_id}/subscribed_apps"
+                params = {"access_token": page_access_token}
+                data = {"subscribed_fields": "messages,messaging_postbacks"}
+
+                response = requests.post(url, data=data, params=params)
+
+                if response.status_code != 200:  # pragma: no cover
+                    raise Exception("Failed to subscribe to app for webhook events")
+
+                self.cleaned_data["page_access_token"] = page_access_token
+                self.cleaned_data["name"] = name
+
+                url = f"https://graph.facebook.com/{page_id}?fields=instagram_business_account"
+                params = {"access_token": page_access_token}
+
+                response = requests.get(url, params=params)
+
+                if response.status_code != 200:  # pragma: no cover
+                    raise Exception("Failed to get IG user")
+
+                response_json = response.json()
+                self.cleaned_data["ig_user_id"] = response_json.get("instagram_business_account").get("id")
+
+            except Exception:
+                raise forms.ValidationError(_("Sorry your Instagram channel could not be connected. Please try again"))
+
+            return self.cleaned_data
+
+    form_class = Form
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["claim_url"] = reverse("channels.types.instagram.claim")
+        context["facebook_app_id"] = settings.FACEBOOK_APPLICATION_ID
+        return context
+
+    def form_valid(self, form):
+        org = self.request.user.get_org()
+        page_id = form.cleaned_data["page_id"]
+        page_access_token = form.cleaned_data["page_access_token"]
+        name = form.cleaned_data["name"]
+        ig_user_id = form.cleaned_data["ig_user_id"]
+
+        config = {Channel.CONFIG_AUTH_TOKEN: page_access_token, Channel.CONFIG_PAGE_NAME: name, "page_id": page_id}
+
+        self.object = Channel.create(
+            org, self.request.user, None, self.channel_type, name=name, address=ig_user_id, config=config
+        )
+
+        return super().form_valid(form)
+
+
+class RefreshToken(ModalMixin, OrgObjPermsMixin, SmartModelActionView):
+    class Form(forms.Form):
+        user_access_token = forms.CharField(min_length=32, required=True, help_text=_("The User Access Token"))
+        fb_user_id = forms.CharField(
+            required=True, help_text=_("The Facebook User ID of the admin that connected the channel")
+        )
+
+    slug_url_kwarg = "uuid"
+    success_url = "uuid@channels.channel_read"
+    form_class = Form
+    permission = "channels.channel_claim"
+    fields = ()
+    template_name = "channels/types/instagram/refresh_token.html"
+    title = _("Reconnect Facebook Page")
```

Comment:
@Robi9 I really think this title should not include Facebook, I think having Instagram there is better to not confuse this view with the other Facebook type

Please share screenshots for the the different screens that way we'll see easily the text that could be confusing